docs: user guide about graceful pod termination

viktor-f opened the pull request:
⚠️ IMPORTANT ⚠️: This is a public repository. Make sure to not disclose:

- [ ] personal data beyond what is necessary for interacting with this Pull Request;
- [ ] business confidential information, such as customer names.

Quality gates:

- [x] I'm aware of the [Contributor Guide](../CONTRIBUTING.md) and did my best to follow the guidelines.
- [x] I'm aware of the [Glossary](../docs/glossary.md) and did my best to use those terms.

Fixes #906

This writes about graceful termination of pods and how users can manage that.

simonklb commented on docs/user-guide/prepare-application.md:
```diff
@@ -39,6 +39,21 @@ git clone https://github.com/elastisys/compliantkubernetes/
+
+1. Something triggers the Pod termination
+1. Any `preStop` [hooks](https://kubernetes.io/docs/concepts/containers/container-lifecycle-hooks/) in the Pod are triggered.
+1. TERM signal is sent to each Container in the Pod.
```
SIGTERM and SIGKILL or TERM and KILL to make it coherent?

viktor-f replied:
Thanks, thought I had changed all of them. Fixed now.

simonklb commented on docs/user-guide/prepare-application.md:
```diff
@@ -39,6 +39,21 @@ git clone https://github.com/elastisys/compliantkubernetes/
+1. Something triggers the Pod termination
+1. Any `preStop` [hooks](https://kubernetes.io/docs/concepts/containers/container-lifecycle-hooks/) in the Pod are triggered.
+1. TERM signal is sent to each Container in the Pod.
+1. If the Pod has not terminated gracefully within the grace period, then SIGKILL is sent to all processes in the Pod.
```
```suggestion
1. If the `preStop` hook or the Pod has not terminated gracefully within the grace period, then SIGKILL is sent to all processes in the Pod.
```

simonklb reviewed (approved):
Just some minor things + fix one sentence per line. Other than that, LGTM!

viktor-f commented:
Updated now. But I'm not sure if we are trying to follow "one sentence per line" in this repo, since there are a lot of pages that does not do that. I changed it regardless since I agree that it is nice to have.

cristiklein reviewed (approved):
Nice addition, thanks! Indeed, getting Pod termination right is something new when coming to the Kubernetes world.

I'd also like us to go for one-sentence-per-line. I know this is not consistent in this repo, but that's where we will slowly head to.
